bpo-44340: Update whatsnews for ThinLTO (#28229)
python/cpython: Doc/whatsnew/3.11.rst

Build Changes
=============

> * CPython is now able to build with the ThinLTO option via `--with-lto=thin`.

Reviewer: ```suggestion
* CPython can now be built with the ThinLTO option via ``--with-lto=thin``.
```
